pkg/common/cns-lib/volume/manager.go

```diff
@@ -973,9 +973,22 @@ func (m *defaultManager) DetachVolume(ctx context.Context, vm *cnsvsphere.Virtua
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
 			faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+
+			fault, isManagedObjectNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.ManagedObjectNotFound)
+			if isManagedObjectNotFoundFault && fault.Obj.Type == cnsDetachSpec.Vm.Type &&
```

Comment:
can we add a nil check for fault.Obj

Reply:
fault.Obj is not a pointer, it is of type vim25types.ManagedObjectReference, hence couldn't add nil check